src/Illuminate/Http/Request.php

```diff
@@ -255,6 +255,22 @@ public function has($key)
 	}
 
 	/**
+	 * Determine if the request contains a non-empty value for an input item matching a given pattern.
+	 *
+	 * @param  string  $pattern
+	 * @return bool
+	 */
+	public function hasRegex($pattern)
+	{
+		foreach ($this->all() as $key => $value)
+		{
+			if (preg_match($pattern, $key) && ! $this->isEmptyString($key)) return true;
```

Comment:
You should be checking for an empty string first, since it's a much cheaper operation than running a regex.

Then squash.